docs/architecture/ErrorReporting.md

```diff
@@ -0,0 +1,144 @@
+# Handling and reporting errors
+
+    WARNING: This document describes future plans for error handling.
+    The UI library does not (yet) implement error handling as described here.
+
+This document describes how errors are handled within the ACS UI library and how they are reported to the surrounding application.
+
+## Error sources
+
+There are two categories of errors, and they are handled differently.
+
```

Comment:
What if hypothetically we introduced a way to directly upload files to azure file storage for chat messages, would errors from that be an ACS backend error? (or a new category if we were to implement it?)